### PR TITLE
Fix upload problems for AUTH OAT

### PR DESCRIPTION
**Pull request by jantje**
This is part one as I also need to look at ssh yun
But as these are functionally not related I want to push this part already.

### CODE REVIEW

File: io.sloeber.core/src/io/sloeber/core/tools/uploaders/arduinoUploader.java [hunk 1]
```diff
@@ -111,20 +104,16 @@ public boolean uploadUsingPreferences(IFile hexFile, boolean usingProgrammer, IP
 	 */
 	@SuppressWarnings("nls")
 	private String getPasswordFromCode() {
-		return IndexHelper.findParameterInFunction(this.myProject, "setup", "ArduinoOTA.setPassword",
+		String parameter = IndexHelper.findParameterInFunction(this.myProject, "setup", "ArduinoOTA.setPassword",
```
Comment (FrzMe) [inline io.sloeber.core/src/io/sloeber/core/tools/uploaders/arduinoUploader.java]: But why?! I just spent one hour trying to figure out why it was not using the password I entered in the port settings. In my code I had a variable which contains the password - it used the variable name as password which did not work.

This is not very user friendly. Please allow us to just specify the password when uploading. Also why forbid changing the password via OTA?

Comment (jantje) [inline io.sloeber.core/src/io/sloeber/core/tools/uploaders/arduinoUploader.java]: I think grabbing the password from the code is a better way. I may be wrong

Comment (FrzMe) [inline io.sloeber.core/src/io/sloeber/core/tools/uploaders/arduinoUploader.java]: It has a lot of shortcomings (doesn't always work, can't change password, intransparent what's going on/why it's not working) therefore I believe offering an alternative would be nice

Comment (jantje) [inline io.sloeber.core/src/io/sloeber/core/tools/uploaders/arduinoUploader.java]: feel free

**Pull request merged**